dotnet/src/Extensions/PromptTemplates.Handlebars/HandlebarsPromptTemplate.cs

```diff
@@ -96,10 +96,12 @@ private KernelArguments GetVariables(KernelArguments? arguments)
 
         foreach (var p in this._promptModel.InputVariables)
         {
-            if (!string.IsNullOrEmpty(p.Default))
+            if (p.Default == null || (p.Default is string stringDefault && string.IsNullOrEmpty(stringDefault)))
```

Comment:
Nit: by the time we get to the IsNullOrEmpty check, we know it's not null, so it could just be a length check

Reply:
Should we also consider whitespaces?

Reply:
Thanks, fixed.